nixos/modules/services/web-apps/dolibarr/default.nix

```diff
@@ -0,0 +1,237 @@
+        "PATH" = lib.makeBinPath [ myphp ];
+      };
+    };
+    services.nginx = {
+      enable = true;
+      virtualHosts."${app}.com" = {
+        listen = [{
+          addr = "0.0.0.0";
+          port = 80;
+        }];
+        root = "${webroot}";
+        locations."/" = {
+          tryFiles = "$uri $uri/ $uri.php";
+          index = "index.php";
+        };
+        locations."~ \\.php$" = {
+          tryFiles = "$uri =404";
+          fastcgiParams = {
+            SCRIPT_FILENAME = "$document_root$fastcgi_script_name";
+            SERVER_NAME = "$http_host";
+            PHP_VALUE = "upload_max_filesize=5M \\n post_max_size=5M";
+          };
+          extraConfig = ''
+            fastcgi_pass             unix:${config.services.phpfpm.pools.${app}.socket};
+            fastcgi_index            index.php;
+            include                  ${pkgs.nginx}/conf/fastcgi_params;
+            include                  ${pkgs.nginx}/conf/fastcgi.conf;
+            fastcgi_intercept_errors on;
+            fastcgi_split_path_info  ^(.+\.php)(.+)$;
+          '';
+        };
+      };
+    };
+
+    systemd.services.dolibarr-mysql-db-init = {
+      description = "Initialize mysql db for ${app}";
+      wants = [ "mysql.service" ];
+      after = [ "mysql.service" ];
+      serviceConfig = let
+        script =  let
+          mysqlPasswordActivation = let
+              initialScript = ''
+                CREATE USER IF NOT EXISTS '${app}'@'localhost' IDENTIFIED BY '${passwordPlaceholder}';
+                GRANT ALL PRIVILEGES ON ${app}.* TO '${app}'@'localhost';
+              '';
+            in (if (cfg.initialDbPasswordFile != null) then ''
+              tmp_script_file=$(mktemp)
+              echo ${escapeShellArg initialScript} > $tmp_script_file
+              ${pkgs.replace-secret}/bin/replace-secret '${passwordPlaceholder}' '${cfg.initialDbPasswordFile}' $tmp_script_file
+              cat $tmp_script_file | ${config.services.mysql.package}/bin/mysql -u root -N
+              rm -f $tmp_script_file
+            '' else "");
+        in ''
+          #!${pkgs.runtimeShell}
+          # Setup database password
+          echo 'SELECT User FROM mysql.user WHERE User = "${app}";' | ${config.services.mysql.package}/bin/mysql -u root -N | grep '${app}'
+          if [ ! $? = 0 ]; then
+            ${mysqlPasswordActivation}
+          fi
+        '';
+        startScript = pkgs.writeScriptBin "${app}-mysql-db-init-start" script;
+      in {
+        Type = "oneshot";
+        ExecStart = "${startScript}/bin/${app}-mysql-db-init-start";
+      };
+    };
+
+    systemd.services.nginx = {
+      wants = [ "${app}-mysql-db-init.service" ];
+      after = [ "${app}-mysql-db-init.service" ];
+    };
+    systemd.services.phpfpm-dolibarr = {
+      serviceConfig.ReadWriteDirectories = [
+        "/etc/${app}"
+        "/var/lib/${app}/documents"
+      ];
+    };
+
+    system.activationScripts.dolibarr-init-script = let
```

Comment:
Is there a good reason to run this as an activation script rather than as a service init script?